### ISSUE TITLE
`vitest/expect-expect` should allow chai style `assert`s

### ISSUE THREAD

**Issue by scarf005** [vitest-dev/eslint-plugin-vitest#586]
(continued from #345 because it was closed without being resolved)

the lint rule only accepting jest style `expect` api is problematic because `expect(foo).toBeDefined()` doesn't assert foo to be defined, wheras chai style `assert(foo)` asserts foo to be defined. thus in this example:

```ts
test("foo", () => {
  declare const foo: { bar: string } | undefined
  expect(foo).toBeDefined()
  expect(foo.bar).toBe("baz") // error: foo can be undefined
})

test("foo", () => {
  declare const foo: { bar: string } | undefined
  assert(foo)
  expect(foo.bar).toBe("baz") // OK
})
```

therefore, i believe eslint-plugin-vitest should either

- have a new rule `expect-assert` that **explicitly requires chai-style asserts**
- make `expect-expect` accept chai-style `assert`

**Comment by veritem** [comment]
on 
> make expect-expect accept chai-style assert

I agree, and a PR is welcome!

**Comment by scarf005** [comment]
my bad; i just checked `@vitest/eslint-plugin` and it seems to handle both `expect` and `assert`. looks like was the issue was because i was using an old version of elsint (pre 9) + `eslint-plugin-vitest`.
